lib/OpenQA/Shared/Plugin/Gru.pm

```diff
@@ -253,8 +253,16 @@ sub enqueue_git_clones ($self, $clones, $job_ids) {
     return unless OpenQA::App->singleton->config->{'scm git'}->{git_auto_clone} eq 'yes';
     # $clones is a hashref with paths as keys and git urls as values
     # $job_id is used to create entries in a related table (gru_dependencies)
-    my $found = $self->_find_existing_minion_job('git_clone', $clones, $job_ids);
-    $self->enqueue('git_clone', $clones, {priority => 10}, $job_ids) unless $found;
+
+    # resolve all symlinks in keys of $clones to allow _find_existing_minion_job find and skip itentical jobs
+    my $clones_sr = {};
+    for my $path (keys %$clones) {
+        my $path_sr = (-e $path) ? Mojo::File->new($path)->realpath : $path;    # resolve symlinks
```

Comment:
This way there's no extra stat for `-e` and also no race condition (as the existence of the file might change between `-e` and `->realpath`):
```suggestion
        my $path_sr = eval { path($path)->realpath } // $path;
```

I'd also get rid of the comment because it is quite redundant.

Reply:
Hm - if there is the possibility of race conditions because multiple calls of `enqueue_git_clones` could run at the samt time, there would need to be a mutex guard for the whole block:

```
  257     # resolve all symlinks in keys of $clones to allow _find_existing_minion_job find and skip identical jobs
  258     my $clones_sr = {};
  262     for my $path (keys %$clones) {
  263         my $path_sr = eval { path($path)->realpath } // $path;
  264         $clones_sr->{$path_sr} = $clones->{$path};
  266     }
  267
  268     my $found = $self->_find_existing_minion_job('git_clone', $clones_sr, $job_ids);
  269     $self->enqueue('git_clone', $clones_sr, {priority => 10}, $job_ids) unless $found;
```

otherwise the symlink resolving could change behaviour depending on paths existing or not and also the `_find_existing_minion_job` method might race against `enqueue` of another thread.

Is this a scenario that we need to cover?

Reply:
We still have the minion job locking per directory, so two minion jobs with the same args won't be a problem.

Minimizing the number of minion jobs with the same  arguments is just best effort - it can always happen because we don't check for that in the transaction. In fact, I've seen multiple identical jobs on OSD several times in the last days. That seems to happen when jobs are quickly created after each other and all of them don't find an existing job in parallel and then create one.